src/zampy/datasets/fapar_lai.py

```diff
@@ -150,8 +150,7 @@ def load(
         variable_names: list[str],
     ) -> xr.Dataset:
         files = list((ingest_dir / self.name).glob("*.nc"))
-
-        ds = xr.open_mfdataset(files, parallel=True)
+        ds = xr.open_mfdataset(files)
```

Comment:
The parallel kwarg was to ensure the dataset being opened as Dask arrays, to avoid any memory issues. Will that go OK now?

Reply:
if the parallel kwargs is True, the open and preprocess steps will be performed in parallel using dask. So, we need to configure Dask for the number of jobs or CPUs for parallel processing. Otherwise `dask.distributed.Client()` will use all available cores by default and this causes a segmentation fault error. This is the case in `cli.py` where `n_workers` are not set, see [here](https://github.com/EcoExtreML/zampy/blob/4f652d6ee4bdfdf91bfc0a9adffcc3b7e9229392/src/zampy/cli.py#L20).

Reply:
OK. I do think that making use of dask.distributed is quite important for performance. You can configure the defaults with a config file https://docs.dask.org/en/latest/configuration.html#specify-configuration

> `dask.distributed.Client()` will use all available cores by default and this causes a segmentation fault error

I guess on some systems? Which seems like something else should be causing the segfault. On my laptop it's no problem starting as many workers as cpu threads, and it's the default behavior of dask for a reason.

Reply:
Thanks for explaining. I did some tests, and it looks like the Dask configuration didn’t fix the issue. When I added `parallel=True` back, the segmentation fault errors showed up again on macOS and Linux, but not on Windows, see GA workflows below. We only use `parallel=True` for the `fapar_lai.py` and not for others. Do you know why it’s needed? After refactoring the code as below, the tests are passing locally. What do you think? 
```python
client = Client()
ds = xr.open_mfdataset(files, parallel=True)
client.close()
```

Reply:
issue #65 submitted.

Reply:
> Do you know why it’s needed? After refactoring the code as below, the tests are passing locally. What do you think?

Probably sped up the load step, or at least forced dask usage.

If you instead make sure a dask client is always available the loading should also be parallelized.